packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -468,29 +485,35 @@ function commitBeforeMutationEffectsOnFiber(finishedWork: Fiber) {
           }
           instance.__reactInternalSnapshotBeforeUpdate = snapshot;
         }
-        break;
       }
-      case HostRoot: {
+      break;
+    }
+    case HostRoot: {
+      if ((flags & Snapshot) !== NoFlags) {
```

Comment:
I agree this branch should be wrapped in a flag check but I'm not totally confident that something doesn't accidentally rely on it, so maybe leave a TODO instead of changing in this PR, unless it's need for your changes for some reason I can't think of

Reply:
@acdlite just to check my understanding – previously this switch statement was entirely wrapped in the flag check, this PR inlines them within each case so this specific check on L492 is necessary to preserve behavior